cmd/lotus-shed/fip-0036.go

```diff
@@ -0,0 +1,565 @@
+				if err != nil {
+					return xerrors.Errorf("fail to load miner actor: \n", err)
+				}
+
+				info, err := m.Info()
+				if err != nil {
+					return xerrors.Errorf("fail to get miner info: \n", err)
+				}
+
+				ownerId := lookupId(info.Owner)
+				if m, found := ownerMap[ownerId]; found { //add miner id to owner list
+					m = append(m, addr)
+					ownerMap[ownerId] = m
+				} else {
+					n := []address.Address{addr}
+					ownerMap[ownerId] = n
+				}
+
+				workerId := lookupId(info.Worker)
+				if m, found := workerMap[workerId]; found { //add miner id to worker list
+					m = append(m, addr)
+					workerMap[workerId] = m
+				} else {
+					n := []address.Address{addr}
+					workerMap[workerId] = n
+				}
+
+				bal, err := m.AvailableBalance(act.Balance)
+				if err != nil {
+					return err
+				}
+
+				pow, ok, err := powerState.MinerPower(addr)
+				if err != nil {
+					return err
+				}
+
+				if !ok {
+					pow.RawBytePower = big.Zero()
+				}
+
+				minerActorsInfo[addr] = minerBriefInfo{
+					rawBytePower: pow.RawBytePower,
+					// gets added up outside this loop
+					dealPower: big.Zero(),
+					balance:   bal,
+				}
+			}
+
+			return nil
+		})
+
+		if err != nil {
+			return err
+		}
+
+		fmt.Println("iterating over proposals")
+		dps, _ := marketState.Proposals()
+		if err := dps.ForEach(func(dealID abi.DealID, d market.DealProposal) error {
+			clientId := lookupId(d.Client)
+			if cd, found := clientToDealStorage[clientId]; found {
+				clientToDealStorage[clientId] = big.Add(cd, big.NewInt(int64(d.PieceSize)))
+			} else {
+				clientToDealStorage[clientId] = big.NewInt(int64(d.PieceSize))
+			}
+
+			providerId := lookupId(d.Provider)
+			mai, found := minerActorsInfo[providerId]
+
+			if !found {
+				return xerrors.Errorf("didn't find miner %s", providerId)
+			}
+
+			mai.dealPower = big.Add(mai.dealPower, big.NewInt(int64(d.PieceSize)))
+			minerActorsInfo[providerId] = mai
+			return nil
+		}); err != nil {
+			return xerrors.Errorf("fail to get deals")
+		}
+
+		// now tabulate votes
+
+		approveBalance := abi.NewTokenAmount(0)
+		rejectionBalance := abi.NewTokenAmount(0)
+		clientApproveBytes := big.Zero()
+		clientRejectBytes := big.Zero()
+		msigPendingVotes := make(map[address.Address]msigVote) //map[msig ID]msigVote
+		votedMsigs := make(map[address.Address]struct{})
+		votesIncludingMsigs := make(map[address.Address]uint64)
+		fmt.Println("counting account and multisig votes")
+		for signer, v := range votes {
+			signerId := lookupId(signer)
+			//process votes for regular accounts
+			accountActor, err := st.GetActor(signerId)
+			if err != nil {
+				return xerrors.Errorf("fail to get account account for signer: ", err)
+			}
+
+			clientBytes, ok := clientToDealStorage[signerId]
+			if !ok {
+				clientBytes = big.Zero()
+			}
+
+			if v == APPROVE {
+				approveBalance = types.BigAdd(approveBalance, accountActor.Balance)
+				votesIncludingMsigs[signerId] = APPROVE
```

Comment:
why adding here?

Reply:
oka after figuring out the intention of this map I think this should be removed

Reply:
@jennijuju You're gonna have to say more than that :P 

Why should it be removed?

(The purpose is just to have a map that includes votes from msigs for the purpose of checking owner addresses in the next loop)

Reply:
maybe answer to my q is here: https://github.com/filecoin-project/lotus/pull/9387#discussion_r982659482?